pkg/backends/influxdb/flux/parser.go

```diff
@@ -0,0 +1,123 @@
+package flux
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/trickstercache/trickster/v2/pkg/timeseries"
+	"github.com/trickstercache/trickster/v2/pkg/util/timeconv/duration"
+)
+
+type Parser struct {
+	reader io.Reader
+}
+
+func NewParser(reader io.Reader) *Parser {
+	return &Parser{
+		reader: reader,
+	}
+}
+
+func (p *Parser) ParseQuery() (*Query, error) {
+	valid := false
+	r := bufio.NewReader(p.reader)
+	ln := 0
+	q := &Query{}
+	for line, err := r.ReadString('\n'); ; line, err = r.ReadString('\n') {
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			} else {
+				return nil, err
+			}
+		}
+		// Check for line 0 "from"
+		if ln == 0 {
+			if !strings.Contains(line, "from") {
+				return nil, FluxSyntax(line, "flux scripts must begin with from(bucket: ...)")
+			}
+		}
+		// Check for range
+		if strings.Contains(line, "range") {
+			q.Extent, err = parseRangeFilter(line)
+			if err != nil {
+				return nil, err
+			}
+			valid = true
+		}
+		q.Statement += line + " "
```

Comment:
using the io.ReadAll() approach, be sure to just set the q.Statement = string(b) instead of using incremental appends